### PR TITLE
fix: check if preview_branch_refs exists

### PR DESCRIPTION
**Pull request by Hallidayo**
## What kind of change does this PR introduce?

Bug Fix - Local Dev

## What is the current behavior?

Running the development version of Supabase if you go to the main projects dashboard this error is appearing:

Screenshot 2023-11-19 at 17 35 35

## What is the new behavior?

Added a check for the variable and now the page loads without error.

## Additional context

@MildTomato

### CODE REVIEW

File: apps/studio/components/interfaces/Home/ProjectList/ProjectCard.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ const ProjectCard = ({
   const { name, ref: projectRef } = project
   const desc = `${project.cloud_provider} | ${project.region}`
 
-  const isBranchingEnabled = project.preview_branch_refs.length > 0
+  const isBranchingEnabled = project.preview_branch_refs && project.preview_branch_refs.length > 0
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Home/ProjectList/ProjectCard.tsx:28]: @joshenlim i suppose this only makes sense if default project is also counted in the preview_branch_refs array

Comment (joshenlim) [reply apps/studio/components/interfaces/Home/ProjectList/ProjectCard.tsx:28]: yeah you're right - the api route for default project probably doesnt have this param hence why it crashed here

Comment (joshenlim) [inline apps/studio/components/interfaces/Home/ProjectList/ProjectCard.tsx:28]: i'd opt to do it this way tbh

```suggestion
  const isBranchingEnabled = (project?.preview_branch_refs ?? []).length > 0
```

Comment (Hallidayo) [reply apps/studio/components/interfaces/Home/ProjectList/ProjectCard.tsx:28]: Hi @joshenlim - Shall I create a commit for this?

**Comment by Hallidayo** [comment]
Hi @MildTomato & @joshenlim - I've just updated the change to use @joshenlim line.

**Comment by saltcod** [comment]
@Hallidayo a quick conflict there & we should be g2g!

**Comment by alaister** [comment]
Hey @Hallidayo,

I'm gonna go ahead and close this as I believe it's been fixed in another PR! Thanks!

**Pull request closed without merging**